src/test/resources/pacts/publicapi-direct-debit-connector-create-agreement-with-reference.json

```diff
@@ -0,0 +1,168 @@
+{
+  "provider": {
+    "name": "direct-debit-connector"
+  },
+  "consumer": {
+    "name": "publicapi"
+  },
+  "interactions": [
+    {
+      "description": "Create a mandate request with reference",
+      "providerStates": [
+        {
+          "name": "a gateway account with external id exists",
+          "params": {
+            "gateway_account_id": "7959d395-e720-4081-9e8a-a534cf76460a"
+          }
+        }
+      ],
+      "request": {
+        "method": "POST",
+        "path": "/v1/api/accounts/7959d395-e720-4081-9e8a-a534cf76460a/mandates",
+        "body": {
+          "return_url": "https://example.com/return",
+          "agreement_type": "ON_DEMAND",
+          "service_reference": "serviceref123"
+        }
+      },
+      "response": {
+        "status": 201,
+        "headers": {
+          "Content-Type": "application/json",
+          "Location": "http://localhost:1234/v1/api/accounts/7959d395-e720-4081-9e8a-a534cf76460a/mandates/test_mandate_id_xyz"
+        },
+        "body": {
+          "mandate_id": "test_mandate_id_xyz",
+          "mandate_type": "ON_DEMAND",
+          "service_reference": "serviceref123",
```

Comment:
should this have an expectation about `mandate_reference` as well?

Reply:
As discussed this should be implemented in PP-3922